fedimint-core/src/bitcoin_migration.rs

```diff
@@ -50,89 +50,74 @@ pub fn checked_address_to_unchecked_address(
-    bincode::deserialize(
-        &bincode::serialize(&invoice).expect("Failed to serialize bitcoin30 invoice"),
-    )
-    .expect("Failed to convert bitcoin30 invoice to bitcoin32 invoice")
+    lightning_invoice32::Bolt11Invoice::from_str(&invoice.to_string())
+        .expect("Failed to convert bitcoin30 invoice to bitcoin32 invoice")
 }
 
 pub fn bitcoin30_to_bitcoin32_keypair(
     keypair: &bitcoin30::secp256k1::KeyPair,
 ) -> bitcoin::secp256k1::Keypair {
-    bincode::deserialize(
-        &bincode::serialize(&keypair).expect("Failed to serialize bitcoin30 keypair"),
+    bitcoin::secp256k1::Keypair::from_secret_key(
+        bitcoin::secp256k1::SECP256K1,
+        &bitcoin30_to_bitcoin32_secp256k1_secret_key(&keypair.secret_key()),
     )
-    .expect("Failed to convert bitcoin30 keypair to bitcoin32 keypair")
 }
 
 pub fn bitcoin32_to_bitcoin30_keypair(
     keypair: &bitcoin::secp256k1::Keypair,
 ) -> bitcoin30::secp256k1::KeyPair {
-    bincode::deserialize(
-        &bincode::serialize(&keypair).expect("Failed to serialize bitcoin32 keypair"),
+    bitcoin30::secp256k1::KeyPair::from_secret_key(
+        bitcoin30::secp256k1::SECP256K1,
+        &bitcoin32_to_bitcoin30_secp256k1_secret_key(&keypair.secret_key()),
     )
-    .expect("Failed to convert bitcoin32 keypair to bitcoin30 keypair")
 }
 
 pub fn bitcoin30_to_bitcoin32_secp256k1_secret_key(
     secret_key: &bitcoin30::secp256k1::SecretKey,
 ) -> bitcoin::secp256k1::SecretKey {
-    bincode::deserialize(
-        &bincode::serialize(&secret_key)
-            .expect("Failed to serialize bitcoin30 secp256k1 secret key"),
+    bitcoin::secp256k1::SecretKey::from_slice(secret_key.as_ref()).expect(
+        "Failed to convert bitcoin30 secp256k1 secret key to bitcoin32 secp256k1 secret key",
     )
-    .expect("Failed to convert bitcoin30 secp256k1 secret key to bitcoin32 secp256k1 secret key")
 }
 
 pub fn bitcoin32_to_bitcoin30_secp256k1_secret_key(
     secret_key: &bitcoin::secp256k1::SecretKey,
 ) -> bitcoin30::secp256k1::SecretKey {
-    bincode::deserialize(
-        &bincode::serialize(&secret_key)
-            .expect("Failed to serialize bitcoin32 secp256k1 secret key"),
+    bitcoin30::secp256k1::SecretKey::from_slice(secret_key.as_ref()).expect(
+        "Failed to convert bitcoin32 secp256k1 secret key to bitcoin30 secp256k1 secret key",
     )
-    .expect("Failed to convert bitcoin32 secp256k1 secret key to bitcoin30 secp256k1 secret key")
 }
 
 pub fn bitcoin30_to_bitcoin32_secp256k1_pubkey(
     pubkey: &bitcoin30::secp256k1::PublicKey,
 ) -> bitcoin::secp256k1::PublicKey {
-    bincode::deserialize(
-        &bincode::serialize(&pubkey).expect("Failed to serialize bitcoin30 secp256k1 pubkey"),
-    )
-    .expect("Failed to convert bitcoin30 secp256k1 pubkey to bitcoin32 secp256k1 pubkey")
+    bitcoin::secp256k1::PublicKey::from_str(&pubkey.to_string())
+        .expect("Failed to convert bitcoin30 secp256k1 pubkey to bitcoin32 secp256k1 pubkey")
```

Comment:
![Screenshot 2024-10-27 at 09-01-41 PublicKey in secp256k1 - Rust](https://github.com/user-attachments/assets/606d120e-0b67-48da-8ace-952ff001736e)
![Screenshot 2024-10-27 at 09-01-04 PublicKey in secp256k1 - Rust](https://github.com/user-attachments/assets/f3c8d937-b30c-45dc-af6c-f97fd1b05c77)

Reply:
Done. However, `PublicKey::from_byte_array_compressed()` is introduced in secp256k1 v0.30, but bitcoin v0.30 relies on secp256k1 v0.29, so the best way to do this is using `PublicKey::from_slice()` and then calling `.expect()`.